draft-wood-icnrg-tlvencap.mkd

```diff
@@ -282,9 +296,111 @@ Protocol of the Deframer/Decrypter:
+It should not be obvious to a 3rd party observing the outer packets what is control signaling
+and what is data traffic.  Therefore, all control signaling is done within the same
+outer names as data traffic.
+
+Control signaling is done with a normal Inner Packet that pushes data to the other side.
+We use an Interest with an Inner Name of the form shown below,
+where '_direction_' is 'up' from the consumer to producer or 'down' for the producer to consumer.
+This allows each side to maintain its own sequence number space in the 'seqnum'.  This is similar
+to the use of the sequence number in the DTLS record layer.
+
+Like DTLS, ESIC control messages are unreliable, though they are uniquely named.
+
+The payload of the control Interest uses a TLV equivalent of the TLS record format
+for handshake and alert messages.  Application data is never communicated in these
+records, as they use an Inner Packet with a different Inner Name.
+Inside the payload, a TLV type of Alert (21) or Handshake (22) indicates the purpose of
+the TLV value.  One may concatenate multiple records in to one payload.
+
+~~~
+ControlName := ccnx:/localhost/esic/_direction_/SID=sid/SEQNUM=seqnum
+~~~
+
+## ESIC Control Packets
+
+A control packet is a CCNx Interest Inner Packet.  The name of the control packet is
+as above in the /localhost/esic namesapce.  The Payload of the Interest is the actual data.
+
+The ESIC control packet SHOULD be padded out to a length that is statistically indistinguishable
+from other traffic in the given _direction_.
+
+The Payload of the Interest contains a set of TLV records using the normal CCNx TLV encoding.
+The TLV types and values are defined in the next section.
+
+In the 'up' direction from the consumer to the producer, a control packet can be inserted
+into the Interest stream as normal.  The producer may use this extra outer name to return
+its own control message or send a "no-op" back to consume the extra name.
+
+In the 'down' direction from the producer to the consumer, there is no pre-allocated
+outer name available.  The producer can only send the consumer a control message if the
+consumer has outstanding Interests up to the producer.  If there is one or more oustanding
+interests in the outer name space, the producer normally would send a Content Object or Interest Return
+corresponding to some inner name.  In this case, the producer would instead inject a control
+packet Interest in the downstream.  This means the producer is now short one outer Interest in the
+upstream direction.  Therefore, whenever the Deframer/Decrypter sees a control message
+in the downstream direction, it MUST insert an upstream "no-op" packet, padded out to
+statistically undetectable length, to give the producer back a missing name slot.
+
+We allow one one ESIC control packet in one outer packet.  However, we allow multiple
+Alert messages to be encoded in the payload.
```

Comment:
"...so long as it remains indistinguishable from packets in the given _direction_"